test/fluree/db/storage/s3_indexing_test.clj

```diff
@@ -0,0 +1,134 @@
+                                   :request {:Bucket bucket
+                                             :Prefix prefix}})]
+    (:Contents result)))
+
+(defn get-index-files-by-type
+  "Categorize index files by type (root, post, spot, tspo, opst)"
+  [objects]
+  (let [index-files (filter #(str/includes? (:Key %) "/index/") objects)]
+    {:root (filter #(str/includes? (:Key %) "/index/root/") index-files)
+     :post (filter #(str/includes? (:Key %) "/index/post/") index-files)
+     :spot (filter #(str/includes? (:Key %) "/index/spot/") index-files)
+     :tspo (filter #(str/includes? (:Key %) "/index/tspo/") index-files)
+     :opst (filter #(str/includes? (:Key %) "/index/opst/") index-files)
+     :all index-files}))
+
+(deftest ^:integration ^:pending test-s3-indexing
+  (testing "S3 storage with indexing triggers and validation"
+    ;; Marking as pending until LocalStack setup is improved in CI
+    (setup-env)
+    (let [bucket "fluree-test"
+          prefix "indexing"
+          ledger-name (str "index-test-" (System/currentTimeMillis))
+          context {"@vocab" "https://ns.flur.ee/ledger#"
+                   "ex" "http://example.org/ns/"}]
+
+      (testing "Index creation with low thresholds"
+        (let [conn @(fluree/connect-s3
+                     {:s3-bucket bucket
+                      :s3-prefix prefix
+                      :s3-endpoint "http://localhost:4566"
+                      :defaults {:indexing {:reindex-min-bytes 100
+                                            :reindex-max-bytes 1000}}})]
+
+          (testing "Ledger creation and initial transaction"
+            (let [ledger @(fluree/create conn ledger-name)
+                  db0 (fluree/db ledger)
+                  data1 [{"@id" "ex:alice" "ex:name" "Alice" "ex:age" 30}
+                         {"@id" "ex:bob" "ex:name" "Bob" "ex:age" 25}]
+                  db1 @(fluree/stage db0 {"@context" context "insert" data1})]
+
+              (is (some? ledger) "Ledger should be created successfully")
+              (is (some? db1) "Data should stage successfully")
+
+              @(fluree/commit! ledger db1)
+
+              ;; Wait for async indexing
+              (Thread/sleep 3000)
+
+              (let [objects (list-s3-objects bucket (str prefix "/" ledger-name))
+                    commit-files (filter #(str/includes? (:Key %) "/commit/") objects)]
+                (is (>= (count commit-files) 2) "Should have commit files in S3"))))
+
+          (testing "Index generation after threshold exceeded"
+            (let [ledger @(fluree/load conn ledger-name)
+                  db-current (fluree/db ledger)
+                  data2 [{"@id" "ex:carol" "ex:name" "Carol" "ex:age" 35}
+                         {"@id" "ex:dave" "ex:name" "Dave" "ex:age" 40}]
+                  db2 @(fluree/stage db-current {"@context" context "insert" data2})]
+
+              @(fluree/commit! ledger db2)
+
+              ;; Wait for indexing to complete
```

Comment:
Could you also add a FIXME comment here to replace the sleep with a more deterministic solution?